cpp/daal/src/algorithms/dtrees/gbt/regression/gbt_regression_predict_dense_default_batch_impl.i

```diff
@@ -207,39 +291,182 @@ protected:
 //////////////////////////////////////////////////////////////////////////////////////////
 template <typename algorithmFPType, prediction::Method method, CpuType cpu>
 services::Status PredictKernel<algorithmFPType, method, cpu>::compute(services::HostAppIface * pHostApp, const NumericTable * x,
-                                                                      const regression::Model * m, NumericTable * r, size_t nIterations)
+                                                                      const regression::Model * m, NumericTable * r, size_t nIterations,
+                                                                      bool predShapContributions, bool predShapInteractions)
 {
     const daal::algorithms::gbt::regression::internal::ModelImpl * pModel =
         static_cast<const daal::algorithms::gbt::regression::internal::ModelImpl *>(m);
     PredictRegressionTask<algorithmFPType, cpu> task(x, r);
-    return task.run(pModel, nIterations, pHostApp);
+    return task.run(pModel, nIterations, pHostApp, predShapContributions, predShapInteractions);
 }
 
 template <typename algorithmFPType, CpuType cpu>
 services::Status PredictRegressionTask<algorithmFPType, cpu>::run(const gbt::regression::internal::ModelImpl * m, size_t nIterations,
-                                                                  services::HostAppIface * pHostApp)
+                                                                  services::HostAppIface * pHostApp, bool predShapContributions,
+                                                                  bool predShapInteractions)
 {
     DAAL_ASSERT(nIterations || nIterations <= m->size());
-    DAAL_CHECK_MALLOC(this->_featHelper.init(*this->_data));
+    DAAL_CHECK_MALLOC(_featHelper.init(*_data));
     const auto nTreesTotal = (nIterations ? nIterations : m->size());
-    this->_aTree.reset(nTreesTotal);
-    DAAL_CHECK_MALLOC(this->_aTree.get());
-    for (size_t i = 0; i < nTreesTotal; ++i) this->_aTree[i] = m->at(i);
-    return runInternal(pHostApp, this->_res);
+    _aTree.reset(nTreesTotal);
+    DAAL_CHECK_MALLOC(_aTree.get());
+    for (size_t i = 0; i < nTreesTotal; ++i) _aTree[i] = m->at(i);
```

Comment:
Can it be vectorized?

Reply:
Likely, I'm adding `PRAGMA_VECTOR_ALWAYS` - the tests will show if it worked